test/fixtures/iotFleetProvisioning/registerDevice.js

```diff
@@ -0,0 +1,148 @@
+'use strict';
+// NOTE: `aws-iot-device-sdk-v2` is bundled into the deployment package
+// eslint-disable-next-line import/no-unresolved
+const { mqtt, io, iot, iotidentity } = require('aws-iot-device-sdk-v2');
+
+const createDeferredMessageResolver = () => {
+  let resolve;
+  let reject;
+  const resolvedMessage = new Promise((promiseResolve, promiseReject) => {
+    resolve = promiseResolve;
+    reject = promiseReject;
+  });
+  return { resolvedMessage, resolve, reject };
+};
+
+const subscribe = (fn, defferedResolver, options) => {
+  const { resolve, reject } = defferedResolver;
+  const { channelName, isRejectionChannel } = Object.assign(
+    {
+      channelName: 'DefaultChannel',
+      isRejectionChannel: false,
+    },
+    options
+  );
+  const subscription = fn((error, response) => {
+    if (response) {
+      console.log(`${channelName} message: ${JSON.stringify(response)}`);
+    }
+
+    if (error || !response) {
+      console.log(`Error occurred on channel ${channelName}.`);
+      reject(error);
+    } else {
+      isRejectionChannel ? reject(response) : resolve(response);
+    }
+  });
+
+  return subscription;
+};
+
+const executeKeys = async identity => {
+  console.log('Subscribing to CreateKeysAndCertificate Accepted and Rejected topics..');
+
+  const keysSubRequest = {};
+
+  const createKeysAndCertificateDefferedMessageResolver = createDeferredMessageResolver();
+  const createKeysAndCertificateAcceptedChannel = subscribe(
+    identity.subscribeToCreateKeysAndCertificateAccepted.bind(
+      identity,
+      keysSubRequest,
+      mqtt.QoS.AtLeastOnce
+    ),
+    createKeysAndCertificateDefferedMessageResolver,
+    { channelName: 'CreateKeysAndCertificateAcceptedChannel', isRejectionChannel: false }
+  );
+  const createKeysAndCertificateRejectedChannel = subscribe(
+    identity.subscribeToCreateKeysAndCertificateRejected.bind(
+      identity,
+      keysSubRequest,
+      mqtt.QoS.AtLeastOnce
+    ),
+    createKeysAndCertificateDefferedMessageResolver,
+    { channelName: 'CreateKeysAndCertificateRejectedChannel', isRejectionChannel: true }
+  );
+
+  await Promise.all([
+    createKeysAndCertificateAcceptedChannel.subscription,
```

Comment:
Promise is now directly at `createKeysAndCertificateAcceptedChannel`. What's passed above to `Promise.all` is two `undefined` values